### PR TITLE
Add CoreRuntimeError to the error list that causes Tribler shutdown 

### LINKED ISSUE #6521: [main] Tribler gets stuck on shutting down screen
How to reproduce:
1. Go to `notifier.py`
2. Add line `raise Exception('test')` to 
```python
    def notify(self, subject, *args):
        # We have to call the notifier callbacks through call_soon_threadsafe
        # because the notify method could have been called from a non-reactor thread
        raise Exception('test')
        self._loop.call_soon_threadsafe(self._notify, subject, *args)
```

Note: before the grand refactoring this trick did not lead to the stuck.

image

### PR DESCRIPTION
**Pull request by drew2a**
This PR fixes #6521 by introducing CoreRuntimeError and adding this error to the list that causes Tribler shutdown.

### CODE REVIEW

File: src/tribler-gui/tribler_gui/__init__.py [hunk 1]
```diff
@@ -9,6 +9,11 @@
 
 logger = logging.getLogger(__name__)
 
+
+class CoreError(Exception):
+    """This is the base class for exceptions that causes GUI shutdown"""
```
Comment (drew2a) [inline src/tribler-gui/tribler_gui/__init__.py:14]: @kozlovsky I'm not sure that `__init__.py` is the best place for the `CoreError` class. If you have any suggestions feel free to give them to me.

Comment (kozlovsky) [reply src/tribler-gui/tribler_gui/__init__.py:14]: I think we should have `exceptions.py` file in `tribler_gui` similar to `exceptions.py` file in `tribler_core`

File: src/tribler-gui/tribler_gui/tests/test_core_manager.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from unittest.mock import MagicMock, patch
```
Comment (kozlovsky) [inline src/tribler-gui/tribler_gui/tests/test_core_manager.py:1]: **nitpick:** It seems the tests don't use magic methods of mock objects (such as `__iter__`, `__getitem__`, etc.), so the base `Mock` class should be enough.

Comment (kozlovsky) [inline src/tribler-gui/tribler_gui/tests/test_core_manager.py:1]: **praise** good test coverage, as always!

**Review by kozlovsky — CHANGES_REQUESTED** [review]

File: src/tribler-gui/tribler_gui/__init__.py (hunk 1 above)
Comment (drew2a) [reply src/tribler-gui/tribler_gui/__init__.py]: exceptions.py has been added.

File: src/tribler-gui/tribler_gui/tests/test_core_manager.py (hunk 2 above)
Comment (drew2a) [reply src/tribler-gui/tribler_gui/tests/test_core_manager.py:1]: unittest uses `MagicMock` as a default class for mock objects: https://docs.python.org/3/library/unittest.mock.html#the-mock-class

> The patch() decorators makes it easy to temporarily replace classes in a particular module with a Mock object. By default patch() will create a MagicMock for you. You can specify an alternative class of Mock using the new_callable argument to patch().

Comment (drew2a) [reply src/tribler-gui/tribler_gui/tests/test_core_manager.py:1]: 👍

**Comment by drew2a** [comment]
retest this please

**Review by kozlovsky — APPROVED** [review]

**Pull request merged**